### PR TITLE
Bug 1828551 - Add documentation diagrams for AutoPushFeature

### PR DESCRIPTION
**Pull request by jonalmeida**
Added documentation for the surface layer of AutoPush where the `AutoPushFeature` and the push service (Firebase Cloud Messaging typically on Android) interact with a client device and the application server.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1828551

### CODE REVIEW

File: android-components/components/feature/push/README.md [hunk 1]
```diff
@@ -64,7 +64,70 @@ implementation "org.mozilla.components:feature-push:{latest-version}"
...
+    Device->>Provider: Request device registration token
+    Provider-->>Device: Receive device registration token
+    Device->>AutoPush: Send token
+    end
+    rect rgb(191, 223, 255)
+    Note over Device,Server: Creating a push subscription
+    Device->>AutoPush: Request subscription endpoint
+    AutoPush-->>Device: Receive subscription
+    Device->>Server: Send subscription endpoint + public key
+    end
```
Comment (Amejia481) [inline android-components/components/feature/push/README.md:110]: ✨

**Comment by jonalmeida** [comment]
<details>
</details>

**Comment by jonalmeida** [comment]
<details>
</details>

**Comment by jonalmeida** [comment]
I'm going to land this, but feedback is welcome here too and I'll make follow-up edits. Please mention me.

**Pull request merged**